jetty-util/src/main/java/org/eclipse/jetty/util/thread/strategy/EatWhatYouKill.java

```diff
@@ -66,16 +70,39 @@
 {
     private static final Logger LOG = Log.getLogger(EatWhatYouKill.class);
 
-    private enum State { IDLE, PRODUCING, REPRODUCING }
+    private enum State { IDLE, PENDING, PRODUCING, REPRODUCING }
 
-    private final Locker _locker = new Locker();
+    private static String[] __histories;
+    static
+    {
+        if (LOG.isDebugEnabled() || Boolean.getBoolean("org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.HISTORY"))
+        {
+            __histories = new String[4*4];
+            for (State s : State.values())
+            {
+                __histories[s.ordinal()*4+0]=String.format("%s=%s",s,"--");
+                __histories[s.ordinal()*4+1]=String.format("%s=%s",s,"-C");
+                __histories[s.ordinal()*4+2]=String.format("%s=%s",s,"P-");
+                __histories[s.ordinal()*4+3]=String.format("%s=%s",s,"PC");
+            }
+        }
+        else
+        {
+            __histories = null;
+        }
+    }
+    
     private final LongAdder _nonBlocking = new LongAdder();
     private final LongAdder _blocking = new LongAdder();
     private final LongAdder _executed = new LongAdder();
     private final Producer _producer;
     private final Executor _executor;
     private final ReservedThreadExecutor _producers;
     private State _state = State.IDLE;
+    private Thread _producerThread;
+    private int _ancient;
+    private int _history;
```

Comment:
@sbordet I propose that we remove the _ancient and _history fields and the static array of state strings

Reply:
@sbordet looking at this again, I think the JIT will take out all the history stuff, so I think this is good to merge as is!